fil_fungible_token/src/token/mod.rs

```diff
@@ -606,6 +629,62 @@ fn validate_amount<'a>(
     Ok(a)
 }
 
+#[derive(Debug)]
+pub enum TransferResult {
+    Transfer(TransferReturn),
+    TransferFrom(TransferFromReturn),
+}
+
+impl TryFrom<TransferResult> for TransferFromReturn {
+    type Error = TokenError;
+
+    fn try_from(value: TransferResult) -> Result<TransferFromReturn> {
+        match value {
+            TransferResult::TransferFrom(ret) => Ok(ret),
+            TransferResult::Transfer(_) => Err(TokenError::TransferReturn { result: value }),
+        }
+    }
+}
+
+impl TryFrom<TransferResult> for TransferReturn {
+    type Error = TokenError;
+
+    fn try_from(value: TransferResult) -> Result<TransferReturn> {
+        match value {
+            TransferResult::TransferFrom(_) => Err(TokenError::TransferReturn { result: value }),
+            TransferResult::Transfer(ret) => Ok(ret),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum BurnResult {
+    Burn(BurnReturn),
+    BurnFrom(BurnFromReturn),
+}
+
+impl TryFrom<BurnResult> for BurnReturn {
+    type Error = TokenError;
+
+    fn try_from(value: BurnResult) -> Result<Self> {
+        match value {
+            BurnResult::Burn(ret) => Ok(ret),
+            BurnResult::BurnFrom(_) => Err(TokenError::BurnReturn { result: value }),
+        }
+    }
+}
+
+impl TryFrom<BurnResult> for BurnFromReturn {
+    type Error = TokenError;
+
+    fn try_from(value: BurnResult) -> Result<Self> {
+        match value {
+            BurnResult::BurnFrom(ret) => Ok(ret),
+            BurnResult::Burn(_) => Err(TokenError::BurnReturn { result: value }),
+        }
+    }
+}
+
```

Comment:
An alternative to this might be to have separate library level methods for `transfer` and `transfer_from` which return different types that match the FRC interface